test/e2e/tests/test_table.py

```diff
@@ -155,3 +267,391 @@ def test_enable_ttl(self, forum_table):
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Creating two more GSIs
+        # Delete office-per-city and add office-per-country
+        cr["spec"] = {
+            "attributeDefinitions": [
+                {
+                    "attributeName": "OfficeName",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "Rank",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "Country",
+                    "attributeType": "S"
+                },
+            ],
+            "globalSecondaryIndexes": [
+                # deleting offices-per-city GSI
+                new_gsi_dict("office-per-country", "Country", 5),
+            ]
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.gsi_matches([new_gsi_dict("office-per-country", "Country", 5)]),
+            timeout_seconds=60*20,
+            interval_seconds=15,
+        )
+
+    def test_create_update_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        # Creating two more GSIs
+        # Delete office-per-city and add office-per-country
+        cr["spec"] = {
+            "attributeDefinitions": [
+                {
+                    "attributeName": "OfficeName",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "Rank",
+                    "attributeType": "S"
+                },
+                {
+                    "attributeName": "Country",
+                    "attributeType": "S"
+                },
+            ],
+            "globalSecondaryIndexes": [
+                new_gsi_dict("office-per-city", "City", 10), # update op
+                new_gsi_dict("office-per-country", "Country", 5), # new gsi
+            ]
+        }
+
+        # Patch k8s resource
+        k8s.patch_custom_resource(ref, cr)
+
+        table.wait_until(
+            table_name,
+            table.gsi_matches([new_gsi_dict("office-per-city", "City", 10), new_gsi_dict("office-per-country", "Country", 5)]),
+            timeout_seconds=60*20,
+            interval_seconds=15,
+        )
+
+    def test_create_update_gsi(self, office_table):
+        (ref, res) = office_table
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Get CR latest revision
+        cr = k8s.wait_resource_consumed_by_controller(ref)
```

Comment:
above tests are great. I don't see where local secondary indexes are ever tested though? or updated...

Reply:
Same reason here. LSIs are only mentioned during the creation, and the API will reject any non-valid LSIs, checking that the table is active is equivalent to checking the LSIs